kv/replica_slice.go

```diff
@@ -115,11 +119,13 @@ func (rs replicaSlice) SortByCommonAttributePrefix(attrs []string) int {
 			}
 		}
 		if firstNotOrdered == 0 {
-			return bucket
+			break
 		}
+		rs.randPerm(firstNotOrdered, topIndex, maxIndex)
 		topIndex = firstNotOrdered - 1
 	}
-	return len(attrs)
+	rs.randPerm(0, topIndex, maxIndex)
```

Comment:
@tamird,  I have made changes according to your suggestion. I have tried to create a new single commit in this PR. But I  failed. So I have created a new pull request #3041.